### PR TITLE
Sls offline setup

### PR DESCRIPTION
**Pull request by jakepanter**
- using sls offline plugin and dynamodb local instead of Localstack (better performance during development)
- updated README accordingly
- moved seeding process into serverless.yml

### CODE REVIEW

File: hms-api/README.md [hunk 1]
```diff
@@ -40,17 +40,15 @@ in the Handler layer by use of the [handler-wrapper](src/handler/handler-wrapper
...
-|-------------|-------------------------------------------------------------------------------|-------------------------------------|
-| NVM         | https://github.com/nvm-sh/nvm#installing-and-updating                         | Installing and managing Node.js/NPM |
-| Node.js     | https://github.com/nvm-sh/nvm#usage                                           | Writing code                        |
-| NPM         | https://github.com/nvm-sh/nvm#usage                                           | Node package management             |
-| Jest        | `npm install jest --global`                                                   | Running unit tests                  |
-| Serverless  | https://www.serverless.com/framework/docs/getting-started                     | Building and deploying code         |
-| Docker      | https://docs.docker.com/get-docker                                            | Running LocalStack                  |
-| Python3/PIP | https://www.python.org/downloads                                              | Installing LocalStack               |
```
Comment (verzano) [inline hms-api/README.md:51]: For installing Java I would recommend [SDK Man](https://sdkman.io/)

File: hms-api/package.json [hunk 2]
```diff
@@ -4,26 +4,28 @@
...
-    "@typescript-eslint/eslint-plugin": "5.27.0",
-    "@typescript-eslint/parser": "5.27.0",
+    "@types/node": "17.0.35",
+    "@typescript-eslint/eslint-plugin": "5.26.0",
+    "@typescript-eslint/parser": "5.26.0",
     "eslint": "8.16.0",
     "eslint-config-google": "0.14.0",
     "jest": "27.5.1",
-    "serverless-localstack": "0.4.36",
+    "serverless-offline": "^8.8.0",
```
Comment (verzano) [inline hms-api/package.json:19]: We can be explicit instead of using ^

```suggestion
    "serverless-offline": "8.8.0",
```

**Review by LetsPoke — APPROVED** [review]

File: hms-api/package.json [hunk 3]
```diff
@@ -4,26 +4,28 @@
...
     "ts-jest": "27.1.4",
     "ts-node": "10.8.0",
     "typescript": "4.7.2",
     "uuid": "8.3.2"
   },
   "dependencies": {
-    "@aws-sdk/client-dynamodb": "3.100.0"
+    "@aws-sdk/client-dynamodb": "3.95.0",
+    "serverless-dynamodb-client": "0.0.2",
+    "serverless-dynamodb-local": "^0.2.40"
```
Comment (verzano) [inline hms-api/package.json:29]:
```suggestion
    "serverless-dynamodb-local": "0.2.40"
```

also do we need these in production? or could they be just in dev?

File: hms-api/package.json [hunk 4]
```diff
@@ -4,26 +4,28 @@
...
-    "start": "sls deploy --stage local",
+    "start": "sls offline start --stage local",
     "stop": "sls remove --stage local"
   },
   "license": "ISC",
   "devDependencies": {
     "@types/jest": "27.5.0",
-    "@types/node": "17.0.38",
-    "@typescript-eslint/eslint-plugin": "5.27.0",
-    "@typescript-eslint/parser": "5.27.0",
```
[the remark this answers is not in the document]
Comment (jakepanter) [reply hms-api/package.json:15]: upsi, no clue why

File: hms-api/serverless.yml [hunk 5]
```diff
@@ -2,17 +2,36 @@ service: hms-api
...
-    lambda:
-      mountCode: false
-    docker:
-      sudo: false
+  dynamodb:
+    stages:
+      - local
+    start:
+      port: 8000
+      inMemory: true
```
Comment (verzano) [inline hms-api/serverless.yml:14]: having this set to false will allow you to restart with the same data, might be nice :thinking:

File: hms-api/README.md [hunk 6]
```diff
@@ -40,17 +40,15 @@ in the Handler layer by use of the [handler-wrapper](src/handler/handler-wrapper
...
-| Localstack  | https://github.com/localstack/localstack#installing                           | Mocking AWS resources               |
-| AWS CLI     | https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html | Making requests to LocalStack/AWS   |
```
[the remark this answers is not in the document]
Comment (jakepanter) [reply hms-api/README.md:53]: I left that one in the list

File: hms-api/serverless.yml (hunk 5 above)
Comment (verzano) [reply hms-api/serverless.yml:14]: but then `dbPath` should also be specified maybe :thinking:

**Review by verzano — APPROVED** [review]

**Pull request merged**